vault/request_handling.go

```diff
@@ -1209,7 +1217,9 @@ func (c *Core) RegisterAuth(ctx context.Context, tokenTTL time.Duration, path st
 	case logical.TokenTypeService:
 		// Register with the expiration manager
 		if err := c.expiration.RegisterAuth(ctx, &te, auth); err != nil {
-			c.tokenStore.revokeOrphan(ctx, te.ID)
+			if err := c.tokenStore.revokeOrphan(ctx, te.ID); err != nil {
+				c.logger.Warn("failed to clean up token lease", "request_path", path, "error", err)
```

Comment:
Are you referring to the difference between using `te.ID` here and `resp.Auth.ClientToken` above? The revoke orphan call here ultimately points to to the same thing (i.e. the token ID that we're trying to revoke) since the `auth` that we pass into this func is from `resp.Auth` and we set `auth.ClientToken = te.ID` up in L1208. The revocation is not by client token lease ID.

The reason why we're able to use `te.ID` is because we instantiate a new instance of `te` within the func, and in all three revoke orphan cases we're now referencing the token ID that's being created instead of the request's client token.

Reply:
Yes, I was indeed referring to `te.ID` vs `resp.Auth.ClientToken`. If it's all the same variable and call then the messaging is sensible. Thanks.